### PR TITLE
feat: Issuer WASM - Accept Invitation

### LINKED ISSUE #1: Issuer WASM - Accept Invitation

### PR DESCRIPTION
**Pull request by fqutishat**
closes #1

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://raw.githubusercontent.com/trustbloc/edge-agent/master/LICENSE)
 [![Godocs](https://img.shields.io/badge/godoc-reference-blue.svg)](https://godoc.org/github.com/trustbloc/edge-agent)
 
-[![Build Status](https://dev.azure.com/trustbloc/bloc/_apis/build/status/trustbloc.edge-agent?branchName=master)](https://dev.azure.com/trustbloc/bloc/_build/latest?definitionId=20&branchName=master)
+[![Build Status](https://dev.azure.com/trustbloc/edge/_apis/build/status/trustbloc.edge-agent?branchName=master)](https://dev.azure.com/trustbloc/edge/_build/latest?definitionId=29&branchName=master)
```
Comment (troyronda) [inline README.md]: I assume some instructions are coming? :) @fqutishat

Comment (fqutishat) [reply README.md]: yes

**Comment by fqutishat** [comment]
@troyronda  @sandrask  @rolsonquadras  I try to cover as much as possible for unit-test

File: cmd/http-server/main_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// This test fails otherwise. However, this can't be checked by the unit test framework. The *testing.T argument is
+// only there so that this test gets picked up by the framework but otherwise we don't need it.
+func TestWithoutUserAgs(t *testing.T) { //nolint - see above
+	setUpArgs()
+	main()
+}
+
+// Strips out the extra args that the unit test framework adds
+// This allows main() to execute as if it was called directly from the command line
+func setUpArgs() {
```
Comment (troyronda) [inline cmd/http-server/main_test.go]: setup is usually one word.

File: cmd/http-server/startcmd/start.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+const (
+	hostURLFlagName      = "host-url"
+	hostURLFlagShorthand = "u"
+	hostURLFlagUsage     = "Host Name:Port." +
+		" Alternatively, this can be set with the following environment variable: " + hostURLEnvKey
+	hostURLEnvKey = "HTTP_SERVER_HOST_URL"
+
+	wasmPathFlagName      = "wasm-path"
+	wasmPathFlagShorthand = "w"
+	wasmPathFlagUsage     = "Wasm path." +
```
Comment (troyronda) [inline cmd/http-server/startcmd/start.go]: WASM

File: pkg/didexchange/invitation/invitation.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	"fmt"
+	"syscall/js"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/client/didexchange"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+)
+
+// RegisterHandleInvitationJSCallback register handle invitation js callback
+func RegisterHandleInvitationJSCallback(didexchangeClient *didexchange.Client) {
```
Comment (troyronda) [inline pkg/didexchange/invitation/invitation.go]: Would it make sense go have a generic registration method for all JS method injections of a package?

File: pkg/didexchange/invitation/invitation.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	statusCh := make(chan service.StateMsg, 10)
```
Comment (troyronda) [inline pkg/didexchange/invitation/invitation.go]: Any reason for 10? Would it make sense to parameterize?

File: pkg/didexchange/invitation/invitation.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+	statusCh          chan service.StateMsg
+}
+
+func (c *callback) handleInvitation(this js.Value, inputs []js.Value) interface{} {
+	// https://github.com/golang/go/issues/26382
+	go func() {
+		invitationData := js.Global().Get("document").
+			Call("getElementById", inputs[0].String()).Get("value").String()
+		if invitationData == "" {
+			js.Global().Call("alert", "invitation data is empty")
```
Comment (troyronda) [inline pkg/didexchange/invitation/invitation.go]: We will need to refactor (later) to have UI elements instead of alerts.

**Comment by troyronda** [comment]
Every PR will fail checks due to 80% project code coverage.

File: pkg/didexchange/invitation/invitation.go (hunk 6 above)
Comment (fqutishat) [reply pkg/didexchange/invitation/invitation.go]: yes

File: pkg/didexchange/invitation/invitation.go (hunk 5 above)
Comment (fqutishat) [reply pkg/didexchange/invitation/invitation.go]: I will add TODO

**Pull request merged**